### PR TITLE
Allow NooBaa CR to not be reconciled

### PR DESCRIPTION
**Pull request by jarrpa**
Signed-off-by: Jose A. Rivera <jarrpa@redhat.com>

### CODE REVIEW

File: pkg/controller/storagecluster/noobaa_system_reconciler_test.go [hunk 1]
```diff
@@ -121,6 +121,100 @@ func TestEnsureNooBaaSystem(t *testing.T) {
...
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      namespacedName.Name,
+					Namespace: namespacedName.Namespace,
+				},
+				Spec: v1.StorageClusterSpec{
+					MultiCloudGateway: &v1.MultiCloudGatewaySpec{
+						ReconcileStrategy: "ignore",
+					},
+				},
+			},
```
Comment (obnoxxx) [inline pkg/controller/storagecluster/noobaa_system_reconciler_test.go:197]: can we be more explicit here?
```suggestion
			},
			isCreate: false,
```

File: pkg/controller/storagecluster/noobaa_system_reconciler_test.go [hunk 2]
```diff
@@ -121,6 +121,100 @@ func TestEnsureNooBaaSystem(t *testing.T) {
...
+
+	for _, c := range cases {
+		reconciler := getReconciler(t, &v1alpha1.NooBaa{})
+		reconciler.client.Create(context.TODO(), &cephCluster)
+
+		err := reconciler.ensureNoobaaSystem(&c.sc, noobaaReconcileTestLogger)
+		assert.NoError(t, err)
+
+		if c.isCreate {
+			err = reconciler.client.Get(context.TODO(), namespacedName, &v1alpha1.NooBaa{})
```
Comment (obnoxxx) [inline pkg/controller/storagecluster/noobaa_system_reconciler_test.go:209]: Can we put this above the `if c.isCreate` check? (Reducing the blocks to the assert statement)

**Review by obnoxxx — COMMENTED** [review]
Looks good in general. 
Two small suggestions for the test case code.

File: pkg/controller/storagecluster/noobaa_system_reconciler_test.go (hunk 1 above)
Comment (jarrpa) [reply pkg/controller/storagecluster/noobaa_system_reconciler_test.go:197]: Why? It's already false.

File: pkg/controller/storagecluster/noobaa_system_reconciler_test.go (hunk 2 above)
Comment (jarrpa) [reply pkg/controller/storagecluster/noobaa_system_reconciler_test.go:209]: Sure.

File: pkg/controller/storagecluster/noobaa_system_reconciler_test.go (hunk 1 above)
Comment (jarrpa) [reply pkg/controller/storagecluster/noobaa_system_reconciler_test.go:197]: Eh, I don't have the time to argue, done. :stuck_out_tongue:

**Review by obnoxxx — APPROVED** [review]
/lgtm thanks!

**Review by obnoxxx — APPROVED** [review]
/lgtm

**Review by obnoxxx — APPROVED** [review]
/lgtm

**Review by obnoxxx — APPROVED** [review]
/lgtm

**Pull request merged**